drivers/ieee802154/ieee802154_dw1000_regs.h

```diff
@@ -1873,10 +1873,10 @@
  * Map the channel number to the index in the configuration arrays below.
  *                  Channel: na  1  2  3  4  5 na  7
  */
-const uint8_t dwt_ch_to_cfg[] = {0, 0, 1, 2, 3, 4, 0, 5};
+static const uint8_t dwt_ch_to_cfg[] = {0, 0, 1, 2, 3, 4, 0, 5};
```

Comment:
This is not good, now the array is placed in multiple .c files and occupying the space multiple times. Better to put the arrays in a .c file and just reference the values in there.

Reply:
That is right, if they are referred multiple times, there will be multiple copies. I believe these constants are used only by `ieee802154_dw1000.c`. One option is to move all these constants to the `dw1000.c` file but it will look a bit messy; another option is to move them to another header file which will be included by `dw1000.c`.  Otherwise, they have to be declared in the `dw1000.c` when being used if they are put in a new `.c` file.

Reply:
You could create a new .c file for these values, and then just add "extern var..." in this header file. Then no changes are needed in dw1000.c file.

Reply:
Sounds good.

Reply:
I feel it is still not a good way. Look at the following error:
```
zephyr/drivers/ieee802154/ieee802154_dw1000.c:1238:29: note: in expansion of macro 'ARRAY_SIZE'
 1238 |  if (rf_cfg->rx_shr_code >= ARRAY_SIZE(dwt_lde_repc_defs)) {
```
The array `dwt_lde_repc_defs[]` is defined in the header file without its size specified. So, the macro `ARRAY_SIZE` has no way to figure out its size unless the array size is specified in the header file. 

Another issue is that I have to keep the comments in two files. It seems reasonable to remove the comments in the C file but the context will lose. 

So, either adding the array size for all arrays whose size is not defined in the header file; or moving all constants to a new header file which will be only included by dw1000.c. Any suggestions?